data_filter_azure/README.containerregistry.md

```diff
@@ -0,0 +1,108 @@
+# Data Filter Example with Data from Azure table storage
+
+Usually, OPA keeps policies and data in-memory. But in cases where the context data cannot be loaded, there are sevaral options to support. This directory contains a sample where the data and policies are downloaded as bundles using the [OPA's bundle](https://www.openpolicyagent.org/docs/latest/bundles/) feature. The polcies and data are loaded on the fly and enforced by OPA. Currently OPA is configured to download one bundle from a single source. These bundles are stored in Azure container registry which is OCI-compatible registry. You can read more about this in  [the post](https://stevelasker.blog/2019/01/25/cloud-native-artifact-stores-evolve-from-container-registries/). This sample contains a sample server that uses OPA's Get Document API on the bundle that is configured with it. The server also has an API that downloads the bundle from container registry. OPA is configured to download bundle from this HTTP server and enforce policy using it.
+
+The server itself is implemented in Python using Flask. 
+
+## Running the sample
+
+The is only prototype and doesn't include the authentication. The installation and the model is adopted from the existing sample under data_filter_example of this project
+
+## Data and Policy
+
+A resource called "registry" can contain different types of resources with different supported actions against them. Each authenticated user will have the mappings of allowed resources and actions. For example the below document represents the mappings for a registry owned by user 'bob'. The data represents that Bob can read, write test-repo and read/delete test-chart. 
+
+```json
+{
+    "bob": {
+        "repositories": {
+            "repo1": {
+                "actions": ["read", "write"]
+            },
+            "repo2": {
+                "actions": ["*"]
+            }
+        },
+        "charts": {
+            "chart1": {
+                "actions": ["read", "write"]
+            }
+        },
+        "pipelines": {
+            "*": {
+                "actions": ["read"]
+            }
+        }
+    }
+}
+```
+
+## Running the sample
+
+1. Create the bundle with data and polcies. Create the folder structure with corresponding data and policy rego files.
+
+```bash
+export BUNDLE_TAR=registries.tar.gz
+cd opabundle
+tar -czvf $BUNDLE_TAR .
+```
+
+2. Push the bundle to the Azure container registry. Create an [azure container registry](https://azure.microsoft.com/en-us/services/container-registry/)
+
+3. The sample fetchs the container registry keys and login server from environment variables ACR_LOGINSERVER, ACR_USERNAME and ACR_PASSWORD.
```

Comment:
Change this to "Set environment variables ACR_LOGINSERVER, ACR_USERNAME, and ACR_PASSWORD to the container registry keys and login."